### ISSUE TITLE
False positive: non-lowercase-variable-in-function (N806) for a type alias

### ISSUE THREAD

**Issue by Tinche** [astral-sh/ruff#8773]
Howdy, this might be an easy one.

Using PEP 695 type aliases seems to trigger N806.

```python
def f() -> None:
    type MyInt = int
```

Setting aside that type aliases might be less common on the function level, I don't think this should trigger N806.

**Comment by charliermarsh** [comment]
Thanks!

**Comment by Tinche** [comment]
Damn that's fast! Thanks.
